lib/heapwatch.js

```diff
@@ -11,11 +11,12 @@ const ZERO_CUMULATIVE_GC_INTERVAL = {
 const GC_REPORT_INTERVAL = 1000;
 
 class HeapWatch {
-    constructor(conf, logger, statsd) {
+    constructor(conf, logger, statsd, prometheus) {
```

Comment:
Honestly, I don't like this. There should be a more generic way of injecting metrics methods into HeapWatch. A `Metrics` object (or similar) should be given to HeapWatch and have it used. There is really no need for HeapWatch to have to differentiate between statsd and prometheus.

Reply:
I took a stab at what you suggested, but I'm not sure you'll like it any better: https://github.com/shdubsh/service-runner/commit/6cd254afec8415ac3dabe77240764aa4835ececf

Let me know?